bigframes/utils/log_adapter.py

```diff
@@ -0,0 +1,37 @@
+import functools
+
+from loguru import logger
+
+_log_file_path = None
+_logger = logger
+_api_methods = []
+
+
+def class_logger(decorated_cls):
+    """Decorator that adds logging functionality to each method of the class."""
+    for attr_name, attr_value in decorated_cls.__dict__.items():
+        if callable(attr_value):
+            setattr(decorated_cls, attr_name, method_logger(attr_value))
+    return decorated_cls
+
+
+def method_logger(method):
+    """Decorator that adds logging functionality to a method."""
+
+    @functools.wraps(method)
+    def wrapper(*args, **kwargs):
+        api_method_name = str(method.__name__)
+        if not api_method_name.startswith("_"):
```

Comment:
This isn't entirely accurate. We do want to add when it starts and ends with two `__` characters. Those are the "dunder" methods and how Python overrides operators.